### ISSUE TITLE
How to get code coverage when I run end to end test?

### ISSUE THREAD

**Issue by vladimir-k-implemica** [JetBrains/intellij-ui-test-robot#409]
I am using the IntelliJ UI robot for end-to-end plugin tests. However, when I run tests with coverage, it shows 0% coverage even though everything works fine. It creates projects, performs operations with files, and everything else seems to be working correctly. I am wondering how I can make the code coverage work with the IntelliJ UI robot.

**Comment by nizienko** [comment]
Hi, there is a [project](https://github.com/edgafner/GBrowser/blob/main/build.gradle.kts) with Jacoco configured.
I've tried it with my simple project containing UI test and seems it provide realistic coverage
[Here is how](https://github.com/nizienko/property_switcher/commit/d2a2ce6145a8cc90d37039d54b9e11f98989d6d3#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06) I configured Jacoco:
- I added Jacoco plugin and applied it to the `runIdeForUiTests` task
- I created `uiTestsCoverageReport` task and scheduled it to run after `runIdeForUiTests`
- I switched off Jacoco in the test task since UI tests are running in a separate process from the plugin
- I configured `isIncludeNoLocationClasses=true` option
- To trigger the `uiTestsCoverageReport` task we need to finish `runIdeForUiTests`, so I execute Exit action after my tests are finished. I think in a real project it is better to use something like `RunListener::testRunFinished` or kill Idea with a CI step

So in my case I run `runIdeForUiTests`, then `test` which runs UI test, it finishes `runIdeForUiTests` and triggers `uiTestsCoverageReport`. After that I copy the report to [artifacts](https://github.com/nizienko/property_switcher/actions/runs/8139517648) 

Thanks to @Jonatha1983  for the solution
